esp/services/ws_fs/ws_fsService.cpp

```diff
@@ -2014,29 +2016,34 @@ void CFileSprayEx::readAndCheckSpraySourceReq(IEspContext& context, MemoryBuffer
                 continue;
 
             //Parse the file to find/validate possible host/ip and local path.
-            StringBuffer hostInPath, localPath;
-            const char* path = nullptr;
-            if (isHostInSprayPath(file, sourcePlaneReq, dropZone, sourceIPReq, sourceHostEp, localPath, hostInPath))
-                path = localPath.str();
-            else
-                path = file;
-
             //Validate file path.
-            StringBuffer s;
-            if (isAbsolutePath(path))
+            StringBuffer hostInPath, localPath, absPath;
+            const char* path = nullptr;
+            if (isUNCPath(file, sourcePlaneReq, dropZone, sourceIPReq, sourceHostEp, localPath, hostInPath))
             {
+                path = localPath.str();
                 //Based on the tests, the dfuserver only supports the wildcard inside the file name, like '/path/f*'.
                 //The dfuserver throws an error if the wildcard is inside the path, like /p*ath/file.
                 if (!isPathInPlane(dropZone, path))
```

Comment:
It would make more sense if validateUNCPath also did this check.. it's doing other checks and it has the dropZone

Reply:
Yes. Renaming to validateUNCPath makes more sense if it also does this check.

Reply:
ok yes, although it wasn't just doing "is UNC Path" before this proposed change.